client/src/components/layout/ParkShow.js

```diff
@@ -66,8 +66,85 @@ const ParkShow = (props) => {
     getPark();
   }, []);
 
+  const deleteReview = async (review) => {
+    try {
+      const parkId = props.match.params.id;
+      const reviewId = review.id;
+      const response = await fetch(`/api/v1/parks/${parkId}/reviews/${reviewId}`, {
+        method: "DELETE",
+      });
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json();
+          const newErrors = translateServerErrors(body.errors);
+          return setErrors(newErrors);
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`;
+          const error = new Error(errorMessage);
+          throw error;
+        }
+      } else {
+        park.reviews.forEach((existingReview) => {
+          if (review.id === existingReview.id) {
+            let reviewArray = park.reviews;
+            reviewArray.splice(reviewArray.indexOf(existingReview), 1);
+            setPark({
+              ...park,
+              reviews: reviewArray,
+            });
+          }
+        });
+        let sum = 0;
+        park.reviews.forEach((existingReview) => {
+          sum += existingReview.rating;
+        });
+        let average = sum / park.reviews.length;
+        setPark({
+          ...park,
+          averageRating: average,
+        });
```

Comment:
Lot's of great logic here! I would recommend moving this logic heavy section, lines 87-105, to the backend which is better at managing this for us. See my note in your `parkReviewsRotuer`!